pygmt/tests/test_figure.py

```diff
@@ -436,26 +433,16 @@ def test_disable_external_display(self, monkeypatch):
         assert _get_default_display_method() == "none"
 
     @pytest.mark.skipif(not _HAS_IPYTHON, reason="Run when IPython is installed")
-    def test_notebook_display(self, monkeypatch):
+    # Mock IPython.get_ipython() to return an object with a config attribute, so PyGMT
+    # can detect that an IPython kernel is running.
+    @patch("IPython.get_ipython", return_value=Mock(config={"IPKernelApp": True}))
```

Comment:
Haven't played around with `unittest.mock` much yet, but does this `unittest.mock.path` call need to be here on top as a decorator, rather than in the test function below? I find this looks a little unintuitive (even though it is less code) compared to setting up the MockIPython class inside the unit test which makes it clear what is happening.

Reply:
Here are different ways to mock IPython:

1. Define a custom class, as in the current main branch and use `pytest.monkeypatch`
```
class MockIPython:
    def __init__(self):
        self.config = {"IPKernelApp": True}

    # Mock IPython.get_ipython() to return a MockIPython instance.
    mock_ipython = MockIPython()
    monkeypatch.setattr(IPython, "get_ipython", lambda: mock_ipython)
```

2. Use `mock`/`patch` as a decorator
```
@patch("IPython.get_ipython", return_value=Mock(config={"IPKernelApp": True}))
def test_xxxx():
    ...
```

3. Use `mock`/`patch` as a context manager
```
def test_xxxx():
    with patch("IPython.get_ipython", return_value=Mock(config={"IPKernelApp": True})):
        ....
```
Maybe we should use option 3 although it means more indentations?

Reply:
Yes, option 3 sounds good.

Reply:
OK, I've changed it to context manager in d08001c6e46a0a217a1e40f08e19ad2a37225249.